src/Nest/CommonAbstractions/SerializationBehavior/GenericJsonConverters/EpochSecondsDateTimeJsonConverter.cs

```diff
@@ -4,6 +4,53 @@
 
 namespace Nest
 {
+	internal class EpochStringSecondsDateTimeJsonConverter : DateTimeConverterBase
```

Comment:
I can see how having different clock resolution for different APIs *might* make sense and I can see why ML has adopted millisecond precision.

I think the way out of this is to say that all APIs must use the same precision (ideally the finest grained precision - ticks from UNIX epoch?) and then any consumers can round as they see fit to avoid hiding the loss of precision behind the API call.